### DISCUSSION TITLE
I'm having trouble getting Agents to work

### DISCUSSION THREAD

**Discussion by phalcomb** [danny-avila/LibreChat#5132] [Q&A]
I'm currently trying to setup an Agent to test with, I'm using Ollama on the backend with the llama3.2-vision model. When I submit a prompt I get this error message:

![image](https://github.com/user-attachments/assets/054e169d-e1e1-4161-b769-25f6e76fb070)

In the logs I see this entry:

![image](https://github.com/user-attachments/assets/cd77bd76-0671-4630-92af-843e03bf455b)

I've checked all of the connections and everything looks correct, nothing is not responding, wondering if I'm missing something and someone can point me in the right direction. I do have LibreChat behind a reverse proxy, although I've removed the proxy during troubleshooting and still receive the same message. I've scoured the docs as best I could and searched this forum, I found a similar issue which mentioned somethings about JWT tokens and so-on, I've generated and replaced all of the default ones and still encounter the issue.

Thanks in advance for any guidance.

**Comment by phalcomb** [comment]
Just for giggles I modified the docker-compose-override file to include ollama with LibreChat instead of using my own instance, loaded my models, regular chat works as expected. But still getting the same issue with Agents. Turned on debugging and captured this in case it's helpful.

```
LibreChat         | 2024-12-30T18:54:50.808Z debug: [BaseClient] Loading history:
LibreChat         | {
LibreChat         |   conversationId: "1bb8b903-5558-4b56-9be6-057f6c095f5d",
LibreChat         |   parentMessageId: "00000000-0000-0000-0000-000000000000",
LibreChat         | }
LibreChat         | 2024-12-30T18:54:50.993Z debug: [BaseClient] Context Count (1/2)
LibreChat         | {
LibreChat         |   remainingContextTokens: 127478,
LibreChat         |   maxContextTokens: 127500,
LibreChat         | }
LibreChat         | 2024-12-30T18:54:50.994Z debug: [BaseClient] Context Count (2/2)
LibreChat         | {
LibreChat         |   remainingContextTokens: 127478,
LibreChat         |   maxContextTokens: 127500,
LibreChat         | }
LibreChat         | 2024-12-30T18:54:50.994Z debug: [BaseClient] tokenCountMap:
LibreChat         | 2024-12-30T18:54:50.995Z debug: [BaseClient]
LibreChat         | {
LibreChat         |   promptTokens: 22,
LibreChat         |   remainingContextTokens: 127478,
LibreChat         |   payloadSize: 1,
LibreChat         |   maxContextTokens: 127500,
LibreChat         | }
LibreChat         | 2024-12-30T18:54:50.998Z debug: [api/server/controllers/agents/client.js] chatCompletion
LibreChat         | {
LibreChat         |   baseURL: undefined,
LibreChat         |   // 1 payload(s)
LibreChat         |   payload: [{"role":"user","content":"What are the names in Latin for some of the animals in the zoo?"}],
LibreChat         | }
ollama-1          | [GIN] 2024/12/30 - 18:54:51 | 404 |       6.001µs |      172.18.0.1 | POST     "/v1/chat/completions/chat/completions"
LibreChat         | 2024-12-30T18:54:51.091Z error: [api/server/controllers/agents/client.js #sendCompletion] Unhandled error type 404 404 page not found
LibreChat         | 
LibreChat         | Troubleshooting URL: https://js.langchain.com/d... [truncated]
LibreChat         | 2024-12-30T18:54:51.092Z error: [handleAbortError] AI response error; aborting request: 404 404 page not found
LibreChat         | 
LibreChat         | Troubleshooting URL: https://js.langchain.com/docs/troubleshooting/err... [truncated]
LibreChat         | 2024-12-30T18:54:51.105Z debug: [saveConvo] api/app/clients/BaseClient.js - saveMessageToDatabase #saveConvo
LibreChat         | 2024-12-30T18:54:51.110Z debug: [AgentController] Request closed
```

**Reply by Odrec** [reply]
Did you fix this?

**Reply by phalcomb** [reply]
Yes, I found a solution quite awhile ago. I honestly can't remember how I resolved it though.

**Reply by Odrec** [reply]
I found the solution: https://github.com/danny-avila/LibreChat/discussions/5132#discussioncomment-12690398

:)

**Comment by Odrec** [comment]
Just in case someone stumbles upon this error with Ollama and Agents in Librechat, the fix is to use the base url as this: 

`baseURL: "http://your-base-url/v1/"`

instead of this

`baseURL: "http://your-base-url/v1/chat/completions"`

**Reply by ImadSaddik** [reply]
Thanks @Odrec
